src/web/components/SlotBodyEnd.tsx

```diff
@@ -189,11 +198,22 @@ const MemoisedInner = ({
         // because of the way the hook behaves, it'll only ever go from false ->
         // true once.
         if (hasBeenSeen) {
-            // Add a new entry to the view log when we know an Epic is viewed
-            logView(testName);
-            sendOphanEpicEvent('VIEW');
+            const meta = data?.slot?.meta;
+
+            if (meta) {
+                // Add a new entry to the view log when we know an Epic is viewed
+                logView(meta.abTestName);
+                sendOphanEpicEvent('VIEW', meta);
+            }
         }
-    }, [hasBeenSeen]);
+        // At the moment, given the information we currently store in data, it's
+        // fine to add it as a dependency because it won't ever change after
+        // viewing, so we shouldn't ever run the hook multiple times. If we _do_
+        // ever have an implementation where information in data could change
+        // after viewing, we'll need to revisit this. Perhaps it's clearer to
+        // remove data from the dependencies and disable the exhaustive-deps
+        // rule, as above.
+    }, [hasBeenSeen, data]);
```

Comment:
Yeah would much rather disable exhaustive deps for this part and improve the code clarity (and avoid the massive comment too!)

Reply:
I've removed the `data` object from the deps array and updated the comment.